File: vector-distance.py
import math


def euclidean_distance():
    n1 = int(input().strip())
    vector1 = list(map(float, input().strip().split()))
    n2 = int(input().strip())
    vector2 = list(map(float, input().strip().split()))

    distance = math.sqrt(sum((v1 - v2) ** 2 for v1, v2 in zip(vector1, vector2)))
    print(distance)


euclidean_distance()

File: vector-distance.cpp
#include <vector>
using namespace std;

double vectorDistance(vector<float> v1, vector<float> v2) {
    if(v1.size() != v2.size()) {
        return -1.0;
    }
    double sum = 0.0;
    for(int i=0; i<v1.size(); i++) {
        sum += pow(v1[i] - v2[i], 2);
    }
    return sqrt(sum);
}

int main() {
    int n;
    cin >> n;
    
    vector<float> vec1(n), vec2(n);

    for(int i=0; i<n; i++){
        cin >> vec1[i];
    }

    for(int i=0; i<n; i++){
        cin >> vec2[i];
    }

    double distance = vectorDistance(vec1, vec2);
    
    cout << fixed;
    cout.precision(15);
    cout << distance << endl;

    return 0;
}